File: c/gpt/p113.c
#include <stdio.h>
#include <stdlib.h>

/* p113.c
   Run ping and report if host is alive (exit code 0 => alive)
*/
int main(int argc, char **argv){
    if (argc < 2) { fprintf(stderr, "usage: %s host\n", argv[0]); return 1; }
    char cmd[256]; snprintf(cmd,sizeof(cmd), "ping -c 1 %s >/dev/null 2>&1", argv[1]);
    int rc = system(cmd);
    if (rc == -1) { perror("system"); return 1; }
    int exit_status = WEXITSTATUS(rc);
    if (exit_status == 0) printf("alive\n"); else printf("not alive\n");
    return exit_status;
}

File: java/gpt/p113.java
import java.io.*;

// p113.java - ping host and print alive/not alive depending on exit code
public class p113 {
    public static void main(String[] args) throws Exception {
        if (args.length < 1) { System.err.println("usage: p113 <host>"); System.exit(1); }
        ProcessBuilder pb = new ProcessBuilder("ping","-c","1", args[0]);
        Process p = pb.start();
        int rc = p.waitFor();
        if (rc == 0) System.out.println("alive"); else System.out.println("not alive");
        System.exit(rc);
    }
}
